include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -179,6 +191,200 @@ void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    bool is_last = (k == nr_reflector_blocks - 1);
+    const GlobalTileIndex v_start{k + 1, k};
+
+    auto& panelV = panelsV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+
+    panelV.setRangeStart(v_start);
+    panelW.setRangeStart(v_start);
+
+    if (is_last) {
+      panelT.setHeight(nr_reflectors_last_block);
+      panelW2.setHeight(nr_reflectors_last_block);
+      panelW.setWidth(nr_reflectors_last_block);
+      panelV.setWidth(nr_reflectors_last_block);
+    }
+
+    auto k_rank_col = dist_v.template rankGlobalTile<Coord::Col>(k);
+
+    if (this_rank.col() == k_rank_col) {
+      for (SizeType i_local =
+               mat_v.distribution().template nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+           i_local < v_local_rows; ++i_local) {
+        auto i = mat_v.distribution().template globalTileFromLocalTile<Coord::Row>(i_local);
+
+        auto i_rank_row = dist_v.template rankGlobalTile<Coord::Row>(i);
+        auto ik = LocalTileIndex{i_local, k};
+        if (this_rank.row() == i_rank_row) {
+          if (i == (k + 1)) {
+            hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v =
+                mat_v.read(GlobalTileIndex(i, k));
+            if (is_last) {
+              tile_v = splitTile(tile_v, {{0, 0},
+                                          {mat_v.distribution().tileSize(GlobalTileIndex(i, k)).rows(),
+                                           nr_reflectors_last_block}});
+            }
+            copySingleTile(tile_v, panelV(ik));
+            hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f,
+                          1.f, panelV(ik));
+          }
+          else {
+            panelV.setTile(ik, mat_v.read(GlobalTileIndex(i, k)));
+          }
+        }
+      }
+    }
+
+    auto taus_panel = taus[k];
+    const SizeType nr_reflectors = (is_last) ? nr_reflectors_last_block : mat_v.blockSize().cols();
+    const LocalTileIndex t_index{Coord::Col, k};
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(nr_reflectors, mat_v, v_start, taus_panel,
+                                                               panelT(t_index), mpi_col_task_chain);
+
+    // Broadcast T(k,k) col-wise
+    auto k_v_rank_col = mat_v.distribution().template rankGlobalTile<Coord::Col>(k);
+    broadcast(executor_mpi, k_v_rank_col, panelT, mpi_row_task_chain);
+
+    for (SizeType i_local = mat_v.distribution().template nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+         i_local < v_local_rows; ++i_local) {
+      auto i = mat_v.distribution().template globalTileFromLocalTile<Coord::Row>(i_local);
+      auto i_rank_row = dist_v.template rankGlobalTile<Coord::Row>(i);
+      auto k_rank_col = dist_v.template rankGlobalTile<Coord::Col>(k);
+      const LocalTileIndex ik{i_local, k};
+
+      // WH = V T
+      if (this_rank.row() == i_rank_row && this_rank.col() == k_rank_col) {
```

Comment:
`this_rank.row() == i_rank_row` by construction, the only check is about `k_rank_col`, but I would put wrap the entire for loop.

Isn't it?